### DISCUSSION TITLE
Truncated/folded distributions

### DISCUSSION THREAD

**Discussion by sdwfrost** [probcomp/Gen.jl#535] [Q&A] [answered]
I'd like to use folded or truncated distributions in my proposals for Metropolis-Hastings; is this possible with the `@dist` macro, or do I need to provide a new distribution using the `Distributions` API?

**Comment by ztangent** [comment, marked as answer]
There isn't support for this with the `@dist` DSL unfortunately! But you might be able to use the wrapper functionality provided by https://github.com/probcomp/GenDistributions.jl to wrap a truncated distribution from Distributions.jl.

You can also implement your own distribution in Gen, though note that Gen's API for distributions is not the same as the one for Distributions.jl!
